### PR TITLE
ree_fs: allow seeking past of the end of file

### PR DESCRIPTION
**Pull request by lorc**
OP-TEE (or TAs) sometimes try to seek past of the file. This is normal
and is allowed by POSIX. Problem is that Zephyr FS API does not allow
this.

So wee need to mimic POSIX behavior at our side. In case of "read"
operation, we will just report that zero bytes were read, the same as
POSIX `pread` function does. For "write" operation it is a bit harder,
because we need to pad the file with zeroes up to the desired position.

### CODE REVIEW

**Comment by mlohvynenko** [comment]
`Tested by Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>`

File: tee_supplicant/src/ree_fs.c [hunk 1]
```diff
@@ -255,6 +265,41 @@ static int tee_fs_write(size_t num_params, struct tee_param *params)
...
 	sz = 0;
 	rc = fs_seek(file, offset, SEEK_SET);
+	if (rc == -EINVAL) {
+		/*
+		 * OP-TEE tries to seek past the end of file. POSIX
+		 * allows this, but Zephyr - does not. We need to fill
+		 * the file with zeroes to the desired position.
+		 */
+		off_t cur_offset;
+		char zero_buf[8];
```
Comment (firscity) [inline tee_supplicant/src/ree_fs.c:275]: nit question: is there some reasons for exactly 8?

**Review by firscity — APPROVED** [review]
`Reviewed-by: Dmytro Firsov <dmytro_firsov@epam.com>`

File: tee_supplicant/src/ree_fs.c (hunk 1 above)
Comment (lorc) [reply tee_supplicant/src/ree_fs.c:275]: Nope :)

My only concern is that it should be small enough to fit on the stack.

What are your suggestions?

Comment (firscity) [reply tee_supplicant/src/ree_fs.c:275]: I was just curios, maybe OPTEE usually write 8 byte aligned values or smth and it worth to be mentioned for someone who might change it in future. You can ignore it and have my RB)

**Pull request merged**